src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/UserAttributeServiceImpl.java

```diff
@@ -59,4 +60,33 @@ public void addUserAttributesToUser(ProfessionalUser newUser, List<String> userR
 
     }
 
+    public List<UserAttribute> addUserAttributesToSuperUserWithJurisdictions(ProfessionalUser user, List<UserAttribute> attributes, List<String> jurisdictionIds) {
+
+        prdEnumService.findAllPrdEnums().stream()
+                .filter(prdEnum -> isValidEnumType(prdEnum.getPrdEnumId().getEnumType(),jurisdictionIds,prdEnum))
+                .map(prdEnum -> {
+                    PrdEnum e = new PrdEnum(prdEnum.getPrdEnumId(), prdEnum.getEnumName(), prdEnum.getEnumDescription());
+                    return new UserAttribute(user, e);
+                }).forEach(userAttribute -> attributes.add(userAttribute));
+
+        if (!CollectionUtils.isEmpty(attributes)) {
+
+            userAttributeRepository.saveAll(attributes);
+        }
+
+        return attributes;
+
+    }
+
+    private boolean isValidEnumType(String enumType,List<String> jurisdictionIds, PrdEnum prdEnum) {

```

Comment:
I know we worked on this together, but I didn't see it at the time. This conditional method can be simplified down to: PrdEnumType.values().containes(enumType) and put directly in the filter statement.